tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,379 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC
+ */
+
+import { useBackend, useSharedState } from '../../backend';
+import { Window } from '../../layouts';
+import { Box, Button, Collapsible, Divider, Flex, Image, Input, LabeledList, Section, Slider, Stack, Table, Tooltip } from '../../components';
+import { formatTime, truncate } from '../../format';
+import { TableCell, TableRow } from '../../components/Table';
+
+let backgroundPop = "rgba(0,0,0,0.2)"; // The intent is to use this akin to a #DEFINE but if this is foolish yell @ me
+
+/* This code block explains the general flow of the code regarding menu structure
+   This menu is rather big so hopefully it helps navigate the heavy nesting of elements
+
+Window contains:
+- Left section containing collapsibles of buttons
+- Right section containing settings
+
+  Left section contains:
+    - Collapsibles with buttons
+
+    Collapsibles contain:
+      - Buttons which have image/name/tooltip based off own data.
+
+    Buttons contain:
+      - Leftmost: image of item
+      - Middle: vertical stack, settings button on top and ? tooltip on bottom.
+        - settings tooltip NYI
+        - tooltip button contains information on material and time costs
+      - Right: name, up to 3 lines / 30 characters
+
+  Right section contains:
+    - Search bar
+    - Materials list containing material info/control
+
+    Material list elements contain:
+      - Eject button
+      - Priority swap buttons (swap priority of two materials for fabricator use)
+      - Speed control buttons
+      - Repeat toggle button
+      - Ore buying which contains ores by rockbox
+
+      Each rockbox element contains:
+        - List of ores available for purchase
+
+        Each ore available for purchase contains:
+          - Button with ore icon, name, amount, and cost
+*/
+
+const BlueprintTooltip = (props) => {
+  let { blueprint } = props;
+  let requirements = [];
+  for (let i in blueprint.item_names) {
+    requirements.push(<>{blueprint.item_amounts[i]/10} {blueprint.item_names[i]}<br /></>);
+  }
+  return (
+    <>
+      Resource Costs:<br />
+      {requirements} <br />
+      {formatTime(blueprint.time, "Instananeous")}
+    </>
+  );
+};
+
+const BlueprintButton = (props, context) => {
+  const { act } = useBackend(context);
+  const { name, blueprintData } = props;
+
+  return (
+    <Stack width={15} inline backgroundColor={backgroundPop} mx={0.575} mb={0} p={0.5}>
+      <Stack.Item>
+        <Button
+          ellipsis
+          width={12.2}
+          height={5.3}
+          p={0}
+          onClick={() => act("product", { "blueprint_ref": blueprintData.ref })}
+        >
+          <Stack>
+            <Stack.Item
```

Comment:
First item in a Stack shouldn't have margin to the left/top already, do you need to add these? Also I don't think they have padding by default, either?